### ISSUE TITLE
cat file without eol

### ISSUE THREAD

**Issue by worldmind** [xonsh/xonsh#2603]
In bash:
```
worldmind@host:~$ cat file
FILEDATASTRworldmind@host:~$
```
in xonsh
```
worldmind@host ~ master $ cat file
worldmind@host ~ master $
```
we don't see string from file
looks like problem with buffering

**Comment by scopatz** [comment]
hmm weird. Thanks for reporting @worldmind.  I do not see this behaviour on my machine. Can you please post the output of running the `xonfig` command?

**Comment by worldmind** [comment]
@scopatz 
```
worldmind@host ~ $ xonfig 
+------------------+----------------------+
| xonsh            | 0.6.0                |
| Git SHA          | c29819c5             |
| Commit Date      | Nov 22 17:51:52 2017 |
| Python           | 3.5.3                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | None                 |
| shell type       | readline             |
| pygments         | None                 |
| on posix         | True                 |
| on linux         | True                 |
| distro           | debian               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

**Comment by mkcor** [comment]
Hello,

I'm happy to be finally giving xonsh a try!

I'm getting an in-between behaviour, where the file without EOL first displays almost as expected (with a suspicious prompt position):

![noeol_1](https://user-images.githubusercontent.com/2227806/38211505-b0255d10-3688-11e8-9b0a-310854d87df7.png)

Once I press Enter, I lose the output:

![noeol_2](https://user-images.githubusercontent.com/2227806/38211506-b34af7b6-3688-11e8-9853-f7d7bbeed6b1.png)

Fyi,

```
(xonsh) marianne@marianne-thinkpad ~ $ xonfig
+------------------+-----------------+
| xonsh            | 0.6.0           |
| Python           | 3.6.5           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 1.0.15          |
| shell type       | prompt_toolkit  |
| pygments         | 2.2.0           |
| on posix         | True            |
| on linux         | True            |
| distro           | debian          |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

**Comment by gforsyth** [comment]
Hi @mkcor -- we're glad you're trying it out!
I believe the issue you are seeing is the same as #2093 and is specific to the `prompt_toolkit` backend.  Unfortunately we don't have a fix for that one yet... :(

**Comment by gforsyth** [comment]
Heh, or rather, I have forgotten that there is a workaround -- @mkcor you should be able to set the envvar `$XONSH_APPEND_NEWLINE=True` which will force the insertion of a newline if there is not one present to avoid the overdrawing.

**Comment by worldmind** [comment]
@gforsyth with workaround it is work, thanks

**Comment by scopatz** [comment]
I am suprised that the default is `False`.  Seems like we should also be able to check if we are in interactive mode in prompt-toolkit and running in an uncaptured subproc and add the newline.  Or if we don't add it to the output, add it to the display

**Comment by scopatz** [comment]
Potential fix in #2605

**Comment by mkcor** [comment]
Nice!
